builder/dockerfile/copy.go

```diff
@@ -82,13 +82,28 @@ type copier struct {
 }
 
 func copierFromDispatchRequest(req dispatchRequest, download sourceDownloader, imageSource *imageMount) copier {
+	platform := req.builder.platform
+	if platform == nil {
+		// May be nil if not explicitly set in API/dockerfile
+		platform = &specs.Platform{}
+	}
+	if platform.OS == "" {
+		// Default to the dispatch requests operating system if not explicit in API/dockerfile
+		platform.OS = req.state.operatingSystem
+	}
+	if platform.OS == "" {
+		// This is a failsafe just in case. Shouldn't be hit.
+		platform.OS = runtime.GOOS
+	}
+
 	return copier{
 		source:      req.source,
 		pathCache:   req.builder.pathCache,
 		download:    download,
 		imageSource: imageSource,
-		platform:    req.builder.platform,
+		platform:    platform,
```

Comment:
I notice that after the copier is created, `builder.performCopy()` is called; https://github.com/moby/moby/blob/492545e139e7461aac044149a931bb4b2dd48f75/builder/dockerfile/dispatchers.go#L123-L131


Looking at that function, that seems to use `req.builder.platform`, not the value set on the copier.


https://github.com/moby/moby/blob/f0585d04d079d289b7b9cb572f8ee509f6768402/builder/dockerfile/internals.go#L172


The value is actually used, but only for path separators, and not persisted in the copy instruction itself; https://github.com/moby/moby/blob/b711437bbd8596312c962d4189e9ad4d2108c2dc/builder/dockerfile/copy.go#L98-L106

Should a property beaded to `copyInstruction`, and set to this value?

Reply:
@thaJeztah I'll defer to @tonistiigi here. I don't think it matters, but I don't own the builder code. Just fixing a bug...

Reply:
The platform in https://github.com/moby/moby/blob/f0585d04d079d289b7b9cb572f8ee509f6768402/builder/dockerfile/internals.go#L172 is for the target image of `--from` not for the current instruction so may be different.